llvm/lib/Target/AMDGPU/AMDGPUTargetTransformInfo.cpp

```diff
@@ -1123,6 +1137,19 @@ Value *GCNTTIImpl::rewriteIntrinsicWithAddressSpace(IntrinsicInst *II,
   }
 }
 
+InstructionCost GCNTTIImpl::getPHIScalarizationOverhead(Type *ScalarTy,
+                                                        VectorType *VTy) {
+  if (DL.getTypeSizeInBits(ScalarTy) != 8)
+    return 0;
```

Comment:
Specially treating i8 doesn't make sense, look for the multiples of legal type

Reply:
Outdated